### PR TITLE
Add detection for CLSID hijacking. Also add support for checking if a…

### PR DESCRIPTION
**Pull request by Jack-McDowell**
… file was signed by microsoft

addresses #269

### CODE REVIEW

File: BLUESPAWN-client/src/hunt/hunts/HuntT1122.cpp [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+using namespace Registry;
+
+namespace Hunts{
+
+	HuntT1122::HuntT1122() : Hunt(L"T1122 - COM Hijacking"){
+		dwSupportedScans = (DWORD) Aggressiveness::Intensive;
+		dwCategoriesAffected = (DWORD) Category::Configurations;
+		dwSourcesInvolved = (DWORD) DataSource::Registry;
+		dwTacticsUsed = (DWORD) Tactic::Persistence;
```
Comment (ION28) [inline BLUESPAWN-client/src/hunt/hunts/HuntT1122.cpp:21]: Also Tactic::DefensiveEvasion

File: BLUESPAWN-client/src/hunt/hunts/HuntT1122.cpp [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+	}
+
+	int HuntT1122::ScanIntensive(const Scope& scope, Reaction reaction){
+		LOG_INFO("Hunting for " << name << " at level Cursory");
```
Comment (ION28) [inline BLUESPAWN-client/src/hunt/hunts/HuntT1122.cpp:25]: level Intensive

File: BLUESPAWN-client/src/hunt/hunts/HuntT1122.cpp [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+		reaction.BeginHunt(GET_INFO());
+
+		int detections = 0;
+
+		std::map<std::wstring, std::vector<RegistryKey>> files{};
+
+		for(auto key : CheckSubkeys(HKEY_LOCAL_MACHINE, L"SOFTWARE\\Classes\\CLSID", true, true)){
```
Comment (ION28) [inline BLUESPAWN-client/src/hunt/hunts/HuntT1122.cpp:32]: Would it be better to just enumerate everything in HKEY_CLASSES_ROOT\CLSID instead?

File: BLUESPAWN-client/src/hunt/hunts/HuntT1122.cpp [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+			RegistryKey subkey{ key, L"InprocServer32" };
```
Comment (ION28) [inline BLUESPAWN-client/src/hunt/hunts/HuntT1122.cpp:33]: You need to also check for LocalServer32s

Comment (ION28) [reply BLUESPAWN-client/src/hunt/hunts/HuntT1122.cpp:33]: and potentially InprocHandler32 as well

File: BLUESPAWN-client/src/hunt/hunts/HuntT1122.cpp (hunk 3 above)
Comment (Jack-McDowell) [reply BLUESPAWN-client/src/hunt/hunts/HuntT1122.cpp:32]: HKEY_CLASSES_ROOT is a symlink to KEY_LOCAL_MACHINE\Software\Classes. I also want to scan users (and if a WoW64 redirected key exists, that too)

File: BLUESPAWN-client/src/hunt/hunts/HuntT1122.cpp (hunk 1 above)
Comment (Jack-McDowell) [reply BLUESPAWN-client/src/hunt/hunts/HuntT1122.cpp:21]: done

File: BLUESPAWN-client/src/hunt/hunts/HuntT1122.cpp (hunk 2 above)
Comment (Jack-McDowell) [reply BLUESPAWN-client/src/hunt/hunts/HuntT1122.cpp:25]: Done

File: BLUESPAWN-client/src/hunt/hunts/HuntT1122.cpp (hunk 4 above)
Comment (Jack-McDowell) [reply BLUESPAWN-client/src/hunt/hunts/HuntT1122.cpp:33]: https://docs.microsoft.com/en-us/windows/win32/com/clsid-key-hklm
Apparently not just those

Comment (ION28) [reply BLUESPAWN-client/src/hunt/hunts/HuntT1122.cpp:33]: yeah but those are the only file ones for 32bit+. Add the 2 I mentioned then you should be good

**Comment by Jack-McDowell** [comment]
There are a number of false positives presented (because rundll32 and explorer.exe are legitimate), but that's a separate issue, and the hunt itself functions as intended

File: BLUESPAWN-client/src/hunt/hunts/HuntT1004.cpp [hunk 5]
```diff
@@ -44,6 +44,11 @@ namespace Hunts {
 		}
 		for(auto& detection : notifies){
 			reaction.RegistryKeyIdentified(std::make_shared<REGISTRY_DETECTION>(detection));
+			auto filename{ std::get<std::wstring>(detection.data) };
+			auto filepath{ FileSystem::SearchPathExecutable(filename) };
+			if(filepath){
+				reaction.FileIdentified(std::make_shared<FILE_DETECTION>(FileSystem::File{ *filepath }));
```
Comment (ION28) [inline BLUESPAWN-client/src/hunt/hunts/HuntT1004.cpp:50]: add `detections++;` here

**Pull request merged**